src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdator.java

```diff
@@ -40,30 +44,46 @@
     @Autowired
     private IdamFeignClient idamClient;
 
+    @Autowired
+    private IdamService idamService;
+
     @Autowired
     private ValidationService validationService;
 
     @Autowired
     private AuditService auditService;
 
     @Override
-    public Optional<UserProfileResponse> update(UpdateUserProfileData updateUserProfileData, String userId, ResponseSource origin) {
+    public AttributeResponse update(UpdateUserProfileData updateUserProfileData, String userId, String origin) {
 
-        UserProfile userProfile = validationService.validateUpdate(updateUserProfileData, userId);
-
-        if (validationService.isValidForUserDetailUpdate(updateUserProfileData, userProfile)) {
-            idamClient.updateUserDetails(updateUserProfileData, userId);
-        }
+        AttributeResponse attributeResponse = new AttributeResponse(HttpStatus.OK);
+        boolean isExuiUpdate = validationService.isExuiUpdateRequest(origin);
+        ResponseSource source = (StringUtils.isEmpty(origin) || !isExuiUpdate) ? ResponseSource.SYNC : ResponseSource.API;
```

Comment:
Please change this to check against SYNC rather than EXUI. This needs to work for more than one consumer. At present it will fail to update status if any api who is NOT EXUI on-boards in the future